py/objint_mpz.c

```diff
@@ -396,13 +396,13 @@ mp_int_t mp_obj_int_get_truncated(mp_const_obj_t self_in) {
     }
 }
 
-mp_int_t mp_obj_int_get_checked(mp_const_obj_t self_in) {
+STATIC mp_int_t _mp_obj_int_get_checked(mp_const_obj_t self_in, bool (*as_int)(const mpz_t *i, mp_int_t *value)) {
     if (mp_obj_is_small_int(self_in)) {
         return MP_OBJ_SMALL_INT_VALUE(self_in);
```

Comment:
What are the intended semantics of `mp_obj_int_get_uint_checked()` when the arg is a negative small-int?  Since it's "checked" IMO it should raise an exception, since that is what `mpz_as_uint_checked()` will do (if the mpz is negative).  If you don't want an exception then wouldn't `mp_obj_int_get_truncated()` work fine instead of using this function?

Reply:
The intention of using this (in relation to the Linux kernel port) is mainly to catch errors in  arithmetic overflow errors before they propagate badly via the kernel FFI. From my personal experience those were usually overflows and not underflows and that's why I *think* I never encountered this negative small-int problem. That's also why I don't want to use `mp_obj_int_get_truncated`.

So yeah, it should raise an exception. I'll fix it.

Reply:
Fixed.